### PR TITLE
 [Fix] [Bug] [Upgrade]  Resolved no data on the task instance page after upgrading from 1.3.x to 3.1.x

### PR DESCRIPTION
**Pull request by pinkfloyds**
no data on the task instance page after upgrading from 1.3.4 to 3.1.x 

modify the update script to change t_ds_task_instance.task_definition_version default to 1 .
see https://github.com/apache/dolphinscheduler/issues/15497

### CODE REVIEW

File: dolphinscheduler-dao/src/main/resources/sql/upgrade/2.0.0_schema/mysql/dolphinscheduler_ddl.sql [hunk 1]
```diff
@@ -415,7 +415,7 @@ alter table t_ds_process_instance drop dependence_schedule_times;
 
 -- t_ds_task_instance   note: Data migration is not supported
 alter table t_ds_task_instance change process_definition_id task_code bigint(20) NOT NULL COMMENT 'task definition code';
-alter table t_ds_task_instance add task_definition_version int(11) DEFAULT '0' COMMENT 'task definition version' AFTER task_code;
+alter table t_ds_task_instance add task_definition_version int(11) DEFAULT '1' COMMENT 'task definition version' AFTER task_code;
```
Comment (ruanwenjun) [inline dolphinscheduler-dao/src/main/resources/sql/upgrade/2.0.0_schema/mysql/dolphinscheduler_ddl.sql:418]: The default value of t_ds_task_instance is '0', we cannot change this.

Comment (pinkfloyds) [reply dolphinscheduler-dao/src/main/resources/sql/upgrade/2.0.0_schema/mysql/dolphinscheduler_ddl.sql:418]: Or I could add it here ？
`update t_ds_task_instance set task_definition_version ='1';`

Comment (ruanwenjun) [reply dolphinscheduler-dao/src/main/resources/sql/upgrade/2.0.0_schema/mysql/dolphinscheduler_ddl.sql:418]: LGTM only update the `task_definition_version`, but the version in `t_ds_task_definition`, `t_ds_task_definition_log` should start from `0`, after you upgrade there should only contains one version in these two table, this is strange.

Comment (pinkfloyds) [reply dolphinscheduler-dao/src/main/resources/sql/upgrade/2.0.0_schema/mysql/dolphinscheduler_ddl.sql:418]: t_ds_task_inset.task_definition_version is assigned to 1 for the first time, which is not useful when the default value is 0,the data in the table should not have t_ds_task_instance task_definition_version = 0,so `update t_ds_task_instance set task_definition_version ='1'`

Comment (ruanwenjun) [reply dolphinscheduler-dao/src/main/resources/sql/upgrade/2.0.0_schema/mysql/dolphinscheduler_ddl.sql:418]: Oh, I see the code will set the begin version to 0. So set the default version to 1 is the correct way.
Could you please change the default value in the initialize sql.
https://github.com/apache/dolphinscheduler/tree/dev/dolphinscheduler-dao/src/main/resources/sql
And we also need to alter this latest version https://github.com/apache/dolphinscheduler/tree/dev/dolphinscheduler-dao/src/main/resources/sql/upgrade/3.2.1_schema.

File: dolphinscheduler-dao/src/main/resources/sql/upgrade/3.2.1_schema/mysql/dolphinscheduler_ddl_post.sql [hunk 2]
```diff
@@ -14,3 +14,15 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
 */
+alter table t_ds_process_definition alter column `version` set default 1;
+alter table t_ds_process_definition alter column `version` int(11) NOT NULL;
```
Comment (ruanwenjun) [inline dolphinscheduler-dao/src/main/resources/sql/upgrade/3.2.1_schema/mysql/dolphinscheduler_ddl_post.sql]:
```
alter table t_ds_process_definition modify column `version` int(11) NOT NULL default 1;
```

**Review by pinkfloyds — COMMENTED** [review]
done

File: dolphinscheduler-dao/src/main/resources/sql/upgrade/3.2.1_schema/mysql/dolphinscheduler_ddl_post.sql (hunk 2 above)
Comment (pinkfloyds) [reply dolphinscheduler-dao/src/main/resources/sql/upgrade/3.2.1_schema/mysql/dolphinscheduler_ddl_post.sql]: ok,have been modified

**Comment by ruanwenjun** [comment]
image

**Comment by pinkfloyds** [comment]
Prompt to add the ready-to-merge label. Can you add it? I can't find an action to add a label

**Review by ruanwenjun — DISMISSED** [review]
LGTM

**Review by ruanwenjun — APPROVED** [review]

**Pull request merged**